services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusEntityData.kt

```diff
@@ -126,6 +125,23 @@ data class KaivuilmoitusEntityData(
             invoicingCustomer = invoicingCustomer.toLaskutusyhteystieto(customerReference),
             additionalInfo = additionalInfo,
         )
+
+    fun toJohtoselvityshakemusEntityData(): JohtoselvityshakemusEntityData =
+        JohtoselvityshakemusEntityData(
+            applicationType = ApplicationType.CABLE_REPORT,
+            pendingOnClient = pendingOnClient,
+            name = name,
+            postalAddress = areas.toPostalAddress(),
+            constructionWork = constructionWork,
+            maintenanceWork = maintenanceWork,
+            propertyConnectivity = false,
+            emergencyWork = emergencyWork,
+            rockExcavation = rockExcavation,
+            workDescription = workDescription,
+            startTime = startTime,
+            endTime = endTime,
+            areas = areas?.flatMap { it.toJohtoselvitysHakemusalues() },
```

Comment:
Since we're only interested in the geometries, we can avoid using a separate function:
```suggestion
            areas = areas?.flatMap { it.geometries() }?.map { JohtoselvitysHakemusalue("", it) },
```

Reply:
Good call, thanks!